### DISCUSSION TITLE
Post 2.4.100 upgrade from 2.4.90, Dashboard menu : System Windows Security errors

### DISCUSSION THREAD

**Discussion by kvkamin** [Security-Onion-Solutions/securityonion#13585] [2.4] [answered]
### Version

2.4.100

### Installation Method

Security Onion ISO image

### Description

other (please provide detail below)

### Installation Type

Distributed

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

64

### RAM

128G

### Storage for /

455G

### Storage for /nsm

7TB

### Network Traffic Collection

tap

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

Dashboard menu item "System Windows Security" has two sub-menus, "User Logon Information" and "Logon Failed and Account Lockout " both display this error : "Field source.ip is of the wrong type" for several panels.

This menu item worked under 2.4.90 and previous.

### Guidelines

- [X] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by kvkamin** [comment]
Another broken Dashboard.  "Security Onion - Alerts" - two tables have these errors.
Saved field "source.ip" of data view "logs-*" is invalid for use with the "Terms" aggregation. Please select a new field.
Saved field "destination.ip" of data view "logs-*" is invalid for use with the "Terms" aggregation. Please select a new field.

**Comment by kvkamin** [comment]
add the "Security Onion - RDP"
Saved field "source.ip" of data view "logs-*" is invalid for use with the "Terms" aggregation. Please select a new field.
Saved field "destination.ip" of data view "logs-*" is invalid for use with the "Terms" aggregation. Please select a new field.

**Comment by kvkamin** [comment]
sorry for the slow drip here, but Security Onion - SSH, same issue. 

All my favorites!

**Comment by dougburks** [comment]
The team is looking into this.

**Comment by dougburks** [comment]
We've identified the root cause of this issue and are preparing a hotfix that should be released next week. In the meantime, you should be able to use SOC Dashboards instead of Kibana dashboards. 

For folks that haven't updated yet, we recommend holding off until the hotfix is released to avoid any potential issues.

We've updated the blog post with a warning to this effect as well:
https://blog.securityonion.net/2024/08/security-onion-24100-now-available.html

**Comment by dougburks** [comment]
Security Onion 2.4.100 Hotfix 20240903 now available!
https://blog.securityonion.net/2024/09/security-onion-24100-hotfix-20240903.html

**Comment by kvkamin** [comment]
I applied hotfix and executed the two scripts as instructed.  No change to any of the menus.

**Reply by weslambert** [reply]
Please provide the output of the following:

```
for i in so-logs-system.application so-logs-system.security so-logs-system.system; do      
    sudo so-elasticsearch-index-templates-list $i | jq .index_templates[0].index_template.composed_of | grep so-system-mappings;  done
```

**Reply by kvkamin** [reply]
"so-system-mappings"
  "so-system-mappings"
  "so-system-mappings"

**Reply by weslambert** [reply]
What's the output of the following?

```
sudo so-elasticsearch-query .ds-logs-system.*-default* _mapping | jq '.[].mappings' | grep -A10 source
```

**Reply by kvkamin** [reply]
"source": {
      "properties": {
        "ip": {
          "type": "ip"
        }
      }
    },
    "tags": {
      "type": "keyword",
      "ignore_above": 1024
    },
--
    "source": {
      "properties": {
        "address": {
          "type": "keyword",
          "ignore_above": 1024
        },
        "as": {
          "properties": {
            "number": {
              "type": "long"
            },
--
    "source": {
      "properties": {
        "address": {
          "type": "keyword",
          "ignore_above": 1024
        },
        "as": {
          "properties": {
            "number": {
              "type": "long"
            },
--
    "source": {
      "properties": {
        "as": {
          "properties": {
            "number": {
              "type": "long"
            },
            "organization": {
              "properties": {
                "name": {
                  "type": "keyword",
--
            "Resource": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "ResourceAttributes": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "ResourceManager": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "ResponderCookie": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "RestrictedAdminMode": {
              "type": "keyword",
              "ignore_above": 1024
--
    "source": {
      "properties": {
        "as": {
          "properties": {
            "number": {
              "type": "long"
            },
            "organization": {
              "properties": {
                "name": {
                  "type": "keyword",
--
            "Resource": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "ResourceAttributes": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "ResourceManager": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "ResponderCookie": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "RestrictedAdminMode": {
              "type": "keyword",
              "ignore_above": 1024
--
    "source": {
      "properties": {
        "as": {
          "properties": {
            "number": {
              "type": "long"
            },
            "organization": {
              "properties": {
                "name": {
                  "type": "keyword",
--
            "Resource": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "ResourceAttributes": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "ResourceManager": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "ResponderCookie": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "RestrictedAdminMode": {
              "type": "keyword",
              "ignore_above": 1024
--
    "source": {
      "properties": {
        "as": {
          "properties": {
            "number": {
              "type": "long"
            },
            "organization": {
              "properties": {
                "name": {
                  "type": "keyword",
--
            "Resource": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "ResourceAttributes": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "ResourceManager": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "ResponderCookie": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "RestrictedAdminMode": {
              "type": "keyword",
              "ignore_above": 1024
--
    "source": {
      "properties": {
        "as": {
          "properties": {
            "number": {
              "type": "long"
            },
            "organization": {
              "properties": {
                "name": {
                  "type": "keyword",
--
            "Resource": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "ResourceAttributes": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "ResourceManager": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "ResponderCookie": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "RestrictedAdminMode": {
              "type": "keyword",
              "ignore_above": 1024
--
    "source": {
      "properties": {
        "as": {
          "properties": {
            "number": {
              "type": "long"
            },
            "organization": {
              "properties": {
                "name": {
                  "type": "keyword",
--
            "Resource": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "ResourceAttributes": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "ResourceManager": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "ResponderCookie": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "RestrictedAdminMode": {
              "type": "keyword",
              "ignore_above": 1024
--
    "source": {
      "properties": {
        "as": {
          "properties": {
            "number": {
              "type": "long"
            },
            "organization": {
              "properties": {
                "name": {
                  "type": "keyword",
--
            "Resource": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "ResourceAttributes": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "ResourceManager": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "ResponderCookie": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "RestrictedAdminMode": {
              "type": "keyword",
              "ignore_above": 1024
--
    "source": {
      "properties": {
        "as": {
          "properties": {
            "number": {
              "type": "long"
            },
            "organization": {
              "properties": {
                "name": {
                  "type": "keyword",
--
            "Resource": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "ResourceAttributes": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "ResourceManager": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "RestrictedAdminMode": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "RestrictedSidCount": {
              "type": "keyword",
              "ignore_above": 1024
--
    "source": {
      "properties": {
        "ip": {
          "type": "ip"
        }
      }
    },
    "system": {
      "properties": {
        "syslog": {
          "type": "object"
--
    "source": {
      "properties": {
        "ip": {
          "type": "ip"
        }
      }
    },
    "tags": {
      "type": "keyword",
      "ignore_above": 1024
    },

**Reply by weslambert** [reply]
Let's try this instead:

```
for i in $(sudo so-elasticsearch-query .ds-logs-system.*-default* _mapping | jq -r 'keys[]'); do 
  echo "$i";  
  echo "-----";
  sudo so-elasticsearch-query $i/_mapping/field/source.ip | jq .
  echo
done
```

**Reply by kvkamin** [reply]
.ds-logs-system.application-default-2024.08.16-000001
-----
{
  ".ds-logs-system.application-default-2024.08.16-000001": {
    "mappings": {}
  }
}

.ds-logs-system.application-default-2024.09.03-000003
-----
{
  ".ds-logs-system.application-default-2024.09.03-000003": {
    "mappings": {
      "source.ip": {
        "full_name": "source.ip",
        "mapping": {
          "ip": {
            "type": "ip"
          }
        }
      }
    }
  }
}

.ds-logs-system.auth-default-2024.08.15-000001
-----
{
  ".ds-logs-system.auth-default-2024.08.15-000001": {
    "mappings": {
      "source.ip": {
        "full_name": "source.ip",
        "mapping": {
          "ip": {
            "type": "ip"
          }
        }
      }
    }
  }
}

.ds-logs-system.auth-default-2024.08.30-000002
-----
{
  ".ds-logs-system.auth-default-2024.08.30-000002": {
    "mappings": {
      "source.ip": {
        "full_name": "source.ip",
        "mapping": {
          "ip": {
            "type": "ip"
          }
        }
      }
    }
  }
}

.ds-logs-system.security-default-2024.08.16-000001
-----
{
  ".ds-logs-system.security-default-2024.08.16-000001": {
    "mappings": {
      "source.ip": {
        "full_name": "source.ip",
        "mapping": {
          "ip": {
            "type": "ip"
          }
        }
      }
    }
  }
}

.ds-logs-system.security-default-2024.08.19-000002
-----
{
  ".ds-logs-system.security-default-2024.08.19-000002": {
    "mappings": {
      "source.ip": {
        "full_name": "source.ip",
        "mapping": {
          "ip": {
            "type": "ip"
          }
        }
      }
    }
  }
}

.ds-logs-system.security-default-2024.08.22-000003
-----
{
  ".ds-logs-system.security-default-2024.08.22-000003": {
    "mappings": {
      "source.ip": {
        "full_name": "source.ip",
        "mapping": {
          "ip": {
            "type": "ip"
          }
        }
      }
    }
  }
}

.ds-logs-system.security-default-2024.08.25-000004
-----
{
  ".ds-logs-system.security-default-2024.08.25-000004": {
    "mappings": {
      "source.ip": {
        "full_name": "source.ip",
        "mapping": {
          "ip": {
            "type": "ip"
          }
        }
      }
    }
  }
}

.ds-logs-system.security-default-2024.08.28-000005
-----
{
  ".ds-logs-system.security-default-2024.08.28-000005": {
    "mappings": {
      "source.ip": {
        "full_name": "source.ip",
        "mapping": {
          "ip": {
            "type": "ip"
          }
        }
      }
    }
  }
}

.ds-logs-system.security-default-2024.08.30-000006
-----
{
  ".ds-logs-system.security-default-2024.08.30-000006": {
    "mappings": {
      "source.ip": {
        "full_name": "source.ip",
        "mapping": {
          "ip": {
            "type": "keyword",
            "ignore_above": 1024
          }
        }
      }
    }
  }
}

.ds-logs-system.security-default-2024.09.01-000007
-----
{
  ".ds-logs-system.security-default-2024.09.01-000007": {
    "mappings": {
      "source.ip": {
        "full_name": "source.ip",
        "mapping": {
          "ip": {
            "type": "keyword",
            "ignore_above": 1024
          }
        }
      }
    }
  }
}

.ds-logs-system.security-default-2024.09.03-000009
-----
{
  ".ds-logs-system.security-default-2024.09.03-000009": {
    "mappings": {
      "source.ip": {
        "full_name": "source.ip",
        "mapping": {
          "ip": {
            "type": "ip"
          }
        }
      }
    }
  }
}

.ds-logs-system.syslog-default-2024.08.15-000001
-----
{
  ".ds-logs-system.syslog-default-2024.08.15-000001": {
    "mappings": {}
  }
}

.ds-logs-system.syslog-default-2024.08.30-000002
-----
{
  ".ds-logs-system.syslog-default-2024.08.30-000002": {
    "mappings": {
      "source.ip": {
        "full_name": "source.ip",
        "mapping": {
          "ip": {
            "type": "ip"
          }
        }
      }
    }
  }
}

.ds-logs-system.system-default-2024.08.16-000001
-----
{
  ".ds-logs-system.system-default-2024.08.16-000001": {
    "mappings": {}
  }
}

.ds-logs-system.system-default-2024.09.03-000003
-----
{
  ".ds-logs-system.system-default-2024.09.03-000003": {
    "mappings": {
      "source.ip": {
        "full_name": "source.ip",
        "mapping": {
          "ip": {
            "type": "ip"
          }
        }
      }
    }
  }
}

**Reply by weslambert** [reply]
Try deleting all of the indices with `keyword` as the type for `source.ip`, for example `.ds-logs-system.security-default-2024.08.30-000006` and `.ds-logs-system.security-default-2024.09.01-000007`.

You can do this from the CLI with:

`sudo so-elasticsearch-query .ds-logs-system.security-default-2024.09.01-000006 -XDELETE`

**Reply by kvkamin** [reply]
not sure i got them all..  some said "is the write index for data stream [logs-system.syslog-default] and cannot be deleted"

no menu fixes

**Reply by kvkamin** [reply]
so now:
RDP menu only missing Saved field "destination.ip" of data view "logs-*" is invalid for use with the "Terms" aggregation.
Alerts.  same thing
Windows logon, ok, but i havent tried destination.ip
ssh menu, missing both still

**Reply by weslambert** [reply]
It might be easier at this point to look at the data view from `Kibana -> Stack Management -> Data Views -> logs-*` to see what indices have conflicts.

**Reply by kvkamin** [reply]
There are 5 mapping conflicts: 
destination.ip | keyword, ip Conflict |   |   |   |   | Edit |  
metadata.input.beats.host.ip | keyword, ip Conflict |   |   |   |   | Edit |  
recordedfuture.evidence_details | flattened, object Conflict |   |   |   |   | Edit |  
source.geo.location | object, geo_point Conflict |   |   |   |   | Edit |  
version | keyword, object Conflict |   |   |   |   | Edit |

**Reply by kvkamin** [reply]
What to do about conflicts?

**Reply by weslambert** [reply]
It looks like the conflicts are not related to the hot fix, so they are likely do to something else. Could you please share the data types and indices that are conflicting? It might be best with a screenshot vs trying to capture it in text, but if you are able to do so in a manageable form, that is fine.

**Reply by kvkamin** [reply]
![Screenshot from 2024-09-03 17-06-52](https://github.com/user-attachments/assets/c414e474-e3b8-4d85-85a2-dff918d4c0ee)

**Reply by kvkamin** [reply]
in logs-*

**Reply by weslambert** [reply]
Right, but if you actually click the conflict icon, it should show you the comparison for the fields

**Reply by kvkamin** [reply]
![Screenshot from 2024-09-04 08-38-27](https://github.com/user-attachments/assets/53c08ae4-4be6-430e-8296-7d4d8e7a3c4e)

**Reply by kvkamin** [reply]
![Screenshot from 2024-09-04 08-39-21](https://github.com/user-attachments/assets/56a3508a-5f0b-4755-a340-4485bfcd443b)

**Reply by kvkamin** [reply]
![Screenshot from 2024-09-04 08-40-09](https://github.com/user-attachments/assets/4d5c01ab-274e-4090-b72c-7b9948809445)

**Reply by kvkamin** [reply]
![Screenshot from 2024-09-04 08-41-22](https://github.com/user-attachments/assets/d067cc7d-bf2c-479a-9359-ed3dbc2b5e9b)

**Reply by kvkamin** [reply]
![Screenshot from 2024-09-04 08-42-21](https://github.com/user-attachments/assets/34e0f287-c1e1-444f-a07e-077c146ba34c)

**Reply by kvkamin** [reply]
Any progress on a fix?

**Reply by weslambert** [reply]
We are working on a fix for the next release. For now, you could manually add the mapping for destination.ip.

**Reply by kvkamin** [reply, marked as answer]
thanks.  I did manage to fix all the menus I cared about.

**Comment by TFWol** [comment]
I have the same issue (as well as mapping conflicts) after upgrading from 2.4.90 to 2.4.100-20240903. Standalone install in my case.

**Reply by dougburks** [reply]
@TFWol Please make sure that you've performed the steps in the blog:
https://blog.securityonion.net/2024/09/security-onion-24100-hotfix-20240903.html

If you continue to have issues after performing those steps, then please start a separate discussion and provide detailed information about your system, any agents you've deployed or extra data you're ingesting, and what exactly you're seeing.

**Reply by TFWol** [reply]
I should mention that I had performed the steps while reading through this Issue. It's the same as what @kvkamin was seeing except I had upgraded from .90 directly to the .100 hotfixed version.

I'll submit a new issue if the problem persists past the fix [you mentioned](https://github.com/Security-Onion-Solutions/securityonion/discussions/13585?notification_referrer_id=NT_kwDOAIoE3bMxMjE4NzMzNDYxODo5MDQ1MjEz#discussioncomment-10631300) that's being worked on, but in the meantime what was the workaround that ended up with this [comment](https://github.com/Security-Onion-Solutions/securityonion/discussions/13585?notification_referrer_id=NT_kwDOAIoE3bMxMjE4NzMzNDYxODo5MDQ1MjEz#discussioncomment-10638171)?

**Reply by TFWol** [reply]
It looks like [there's traction](https://github.com/Security-Onion-Solutions/securityonion/issues/13725)
